lib/spack/spack/util/windows_registry.py

```
@@ -0,0 +1,324 @@
+    HKEY_USERS = _HKEY_CONSTANT("HKEY_USERS")
+    HKEY_LOCAL_MACHINE = _HKEY_CONSTANT("HKEY_LOCAL_MACHINE")
+    HKEY_CURRENT_CONFIG = _HKEY_CONSTANT("HKEY_CURRENT_CONFIG")
+    HKEY_PERFORMANCE_DATA = _HKEY_CONSTANT("HKEY_PERFORMANCE_DATA")
+
+
+class WindowsRegistryView(object):
+    """
+    Interface to provide access, querying, and searching to Windows registry entries.
+    This class represents a single key entrypoint into the Windows registry
+    and provides an interface to this key's values, its subkeys, and those subkey's values.
+    This class cannot be used to move freely about the registry, only subkeys/values of
+    the root key used to instantiate this class.
+    """
+
+    def __init__(self, key, root_key=HKEY.HKEY_CURRENT_USER):
+        """Constructs a Windows Registry entrypoint to key provided
+        root_key should be an already open root key or an hkey constant if provided
+
+        Args:
+            key (str): registry key to provide root for registry key for this clas
+            root_key: Already open registry key or HKEY constant to provide access into
+                         the Windows registry. Registry access requires an already open key
+                         to get an entrypoint, the HKEY constants are always open, or an already
+                         open key can be used instead.
+        """
+        if not is_windows:
+            raise RuntimeError(
+                "Cannot instantiate Windows Registry class on non Windows platforms"
+            )
+        self.key = key
+        self.root = root_key
+        self._reg = None
+
+    @contextmanager
+    def invalid_reg_ref_error_handler(self):
+        try:
+            try:
+                yield
+            except FileNotFoundError as e:
+                if e.winerror == 2:
+                    tty.debug("Key %s at position %s does not exist" % (self.key, str(self.root)))
+                else:
+                    raise e
+        finally:
```

What is the purpose of this outer `finally`, would that swallow the exception on the line above? If so, why raise the exception? If not, then I'm not sure what the use of the nesting is.

---

Unsure, I think this may have been a rebasing mistake, or maybe I got turned around during implementation, either way, the nesting is unnecessary. Removed.